src/Reader.php

```diff
@@ -393,6 +393,7 @@ private function getReader($import, $filePath, string $readerType = null, string
             Cell::setValueBinder($import);
         }
 
+        $this->temporaryFileFactory->setFileExtension(pathinfo($filePath, PATHINFO_EXTENSION));
```

Comment:
Please pas the file extension through the `make()` and `makeLocal()` methods.

Reply:
Thank you for review, I have fixed it.